### ISSUE TITLE
S3 disk space usage always increasing

### ISSUE THREAD

**Issue by irizzant** [grafana/tempo#1774]
**Describe the bug**
Hi all,
I've Tempo deployed in microservices mode in my Kubernetes cluster and my Minio S3 storage disk space usage is always increasing even if Compactor is set to use 1h of retention.

Here is my config:
```yaml
compactor:
        compaction:
            block_retention: 1h
            compacted_block_retention: 15m
    distributor: {}
    http_api_prefix: ""
    ingester:
        lifecycler:
            ring:
                replication_factor: 3
    memberlist:
        abort_if_cluster_join_fails: false
        bind_port: 7946
        join_members:
            - gossip-ring.tempo.svc.cluster.local.:7946
    metrics_generator:
        storage:
            remote_write:
                - name: remote-write
                  send_exemplars: true
                  url: http://kube-prometheus-stack-prometheus.monitoring.svc.cluster.local:9090/api/v1/write
    metrics_generator_enabled: true
    multitenancy_enabled: true
    overrides:
        per_tenant_override_config: /overrides/overrides.yaml
    querier:
        frontend_worker:
            grpc_client_config:
                max_send_msg_size: 1.34217728e+08
    search_enabled: true
    server:
        grpc_server_max_recv_msg_size: 1.34217728e+08
        grpc_server_max_send_msg_size: 1.34217728e+08
        http_listen_port: 3200
    storage:
        trace:
            azure: {}
            backend: s3
            blocklist_poll: "0"
            cache: memcached
            gcs: {}
            memcached:
                consistent_hash: true
                host: memcached
                service: memcached-client
                timeout: 200ms
            pool:
                queue_depth: 2000
            s3:
                access_key: ${S3_ACCESS_KEY}
                bucket: tempo
                endpoint: minio.minio.svc.cluster.local:9000
                insecure: true
                secret_key: ${S3_SECRET_KEY}
            wal:
                path: /var/tempo/wal
```

**To Reproduce**
1. Deploy Tempo in microservices mode with Minio as S3 storage
2. Set Compactor with 1h of retention
3. Start sending traces to Tempo
4. Observe Minio disk space usage

**Expected behavior**
Disk space usage should decrease with compaction cycles.

**Environment:**
 - Infrastructure: Kubernetes
 - Deployment tool: jsonnet

**Additional Context**

**Comment by joe-elliott** [comment]
By default compactors should delete compacted blocks `1h` after they have been compacted. This is controlled by `compacted_block_retention`: https://grafana.com/docs/tempo/latest/configuration/#compactor

You should see lines like this in your compactors:
```
level=info ts=2022-10-03T12:12:44.5110932Z caller=retention.go:76 msg="deleting block" blockID=f7eca729-e168-41b6-a3bf-7d302d949278 tenantID=1
level=info ts=2022-10-03T12:12:44.641036004Z caller=retention.go:76 msg="deleting block" blockID=9b401afd-5222-49f2-86f6-120ea3d84eb1 tenantID=1
level=info ts=2022-10-03T12:12:44.787605305Z caller=retention.go:76 msg="deleting block" blockID=7d71e887-4185-423c-bd90-71bff61d04e5 tenantID=1
```

**Comment by irizzant** [comment]
Hi @joe-elliott 
as you can see from the above config the parameter is set to 15m:
```yaml
compacted_block_retention: 15m
```

**Comment by ie-pham** [comment]
To clarify, even with the cronjob to clean up, you still don't see any improvement?

**Comment by amitsetty** [comment]
I have also been experiencing an everlong increase in storage even with a lifecycle deleting the multiparts

**Comment by ie-pham** [comment]
@irizzant Could you let us know what your expected receive MB/sec and also the data for this metric?:
`sum(rate(tempo_distributor_bytes_received_total{}[1m])) / 1000 / 1000`

And to check whether the retention blocks has been successfully deleted, could you also provide:
`sum(increase(tempodb_retention_deleted_total{}[5m]))`

**Comment by irizzant** [comment]
@ie-pham 
> To clarify, even with the cronjob to clean up, you still don't see any improvement?

correct, as @amitsetty said we're experiencing this strange behavior even with the cleanup job enabled.

> You only showed a couple of hours worth of data in your chart, is the behavior you are seeing new? If yes, what was the previous volume that you were seeing before? It seems like it is tapering off around 18 Gb, what number were you expecting it to be?

The chart shows more, starting from 13.20-ish (when I enabled the traces export to Tempo) to 17.30-ish (when I disabled them because of the disk space usage) it's 4 hours.

> Could you let us know what your expected receive MB/sec and also the data for this metric?:
sum(rate(tempo_distributor_bytes_received_total{}[1m])) / 1000 / 1000

> And to check whether the retention blocks has been successfully deleted, could you also provide:
sum(increase(tempodb_retention_deleted_total{}[5m]))

I'm going to enable metrics export in Tempo and will let you know the outcome.

**Comment by irizzant** [comment]
@ie-pham as far as I can see ` sum(rate(tempo_distributor_bytes_received_total{}[1m])) / 1000 / 1000` is getting up to 4.5 MB/sec.

About ` sum(increase(tempodb_retention_deleted_total{}[5m]))` I can see a 4.44 in the last hour.

I'm also receiving notifications about tempo compaction errors about:
```
error shipping block to backend, blockID e873b453-bf85-4397-9b13-856c56edb45c: error completing block: error completing multipart upload, object: jboss-production/e873b453-bf85-4397-9b13-856c56edb45c/data.parquet, obj etag: : Your proposed upload is smaller than the minimum allowed object size.
```

**Comment by irizzant** [comment]
@ie-pham 
Here are the screenshots showing the metrics you asked for:
![image](https://user-images.githubusercontent.com/15982257/193855402-280d4c78-95f5-4534-b6f7-c245199c3eee.png)
![image](https://user-images.githubusercontent.com/15982257/193855669-548c24a0-37fd-46de-87dd-d52f8bf90b80.png)

**Comment by ie-pham** [comment]
@irizzant Thank you for providing that information. We believe you are facing a similar issue described [here](https://github.com/grafana/tempo/issues/1769)  

To verify whether or not it is the same issue, could you try increasing `row_group_size_bytes` to ~100MB and see that helps?

**Comment by irizzant** [comment]
@ie-pham I've tried the suggested fix and this is my updated Compactor ConfigMap:
```yaml
tempo.yaml: |
    compactor:
        compaction:
            block_retention: 1h
            chunk_size_bytes: 1.048576e+07
            compacted_block_retention: 15m
        ring:
            kvstore:
                store: memberlist
    distributor: {}
    http_api_prefix: ""
    ingester:
        lifecycler:
            ring:
                replication_factor: 3
    memberlist:
        abort_if_cluster_join_fails: false
        bind_port: 7946
        join_members:
            - gossip-ring.tempo.svc.cluster.local.:7946
    metrics_generator:
        storage:
            remote_write:
                - name: remote-write
                  send_exemplars: true
                  url: http://kube-prometheus-stack-prometheus.monitoring.svc.cluster.local:9090/api/v1/write
    metrics_generator_enabled: true
    multitenancy_enabled: true
    overrides:
        per_tenant_override_config: /overrides/overrides.yaml
    querier:
        frontend_worker:
            grpc_client_config:
                max_send_msg_size: 1.34217728e+08
    search_enabled: true
    server:
        grpc_server_max_recv_msg_size: 1.34217728e+08
        grpc_server_max_send_msg_size: 1.34217728e+08
        http_listen_port: 3200
    storage:
        trace:
            azure: {}
            backend: s3
            block:
                row_group_size_bytes: 1e+08
            blocklist_poll: 5m
            cache: memcached
            gcs: {}
            memcached:
                consistent_hash: true
                host: memcached
                service: memcached-client
                timeout: 200ms
            pool:
                queue_depth: 2000
            s3:
                access_key: ${S3_ACCESS_KEY}
                bucket: tempo
                endpoint: minio.minio.svc.cluster.local:9000
                insecure: true
                secret_key: ${S3_SECRET_KEY}
            wal:
                path: /var/tempo/wal
```

As you can see I added the `row_group_size_bytes: 1e+08` parameter but still I see (less) errors:
```
error shipping block to backend, blockID 60b3c1ae-4494-485f-8486-8728e51de384: error completing block: error completing multipart upload, object: jboss-production/60b3c1ae-4494-485f-8486-8728e51de384/data.parquet, obj etag: : Your proposed upload is smaller than the minimum allowed object size
```

Disk space usage is still increasing at the moment.

**Comment by ie-pham** [comment]
You mentioned that you are seeing less "minimum size" errors, does the disk space usage still increase as the same rate or has it slowed down? And also do you see an increase in this metric? `sum(increase(tempodb_retention_deleted_total{}[5m]))`.

What we're trying to see is if the disk space usage increment is due to successful flushes or the failed flushes that are caused by the "smaller than the minimum allowed object size" error.

**Comment by irizzant** [comment]
@ie-pham I think I spoke early, I see the same behavior with the disk space usage in Minio always increasing.

These errors:
```
error shipping block to backend, blockID 60b3c1ae-4494-485f-8486-8728e51de384: error completing block: error completing multipart upload, object: jboss-production/60b3c1ae-4494-485f-8486-8728e51de384/data.parquet, obj etag: : Your proposed upload is smaller than the minimum allowed object size
```
where a few at the beginning of the space usage increase but then I saw many.

See below the metrics you requested: 

![image](https://user-images.githubusercontent.com/15982257/195270655-cdeca2c9-8de4-4492-b750-da3a9e7ea01b.png)

**Comment by irizzant** [comment]
@ie-pham can you tell me what exactly the metric name is?

**Comment by ie-pham** [comment]
Apologies for the delay in response. 

Could you provide the following to help us further debug this issue?

- Exact hash for the image that you are using as there are definitely some differences in the last few releases
- Could you confirm whether you are using Parquet?
- About an hour's worth of Compactor logs (this will also provide us the block size)
- Configs from GET  `/status/config` as there may be an issue on our side when applying configs

**Comment by irizzant** [comment]
Sorry for my late reply too:
> Exact hash for the image that you are using as there are definitely some differences in the last few releases

maven-repo.sdb.it:18080/grafana/tempo@sha256:5e5ebabd9bf373779e2a4bdf0959c26545df35d6a79125aa0f64b7ad834be49f

> Could you confirm whether you are using Parquet?

Since the logs show:

object: jboss-production/c11198bd-afbe-4339-b9f7-d6ccd016d3f9/**data.parquet**,

I suppose that Parquet is enabled by default.

> About an hour's worth of Compactor logs (this will also provide us the block size)
[Explore-logs-2022-10-19 20 47 09.txt](https://github.com/grafana/tempo/files/9823643/Explore-logs-2022-10-19.20.47.09.txt)

Sorry I didn't find a way to sort the logs by time.

> Configs from GET /status/config as there may be an issue on our side when applying configs
```yaml
target: compactor
multitenancy_enabled: true
search_enabled: true
metrics_generator_enabled: true
http_api_prefix: ""
server:
    http_listen_network: tcp
    http_listen_address: ""
    http_listen_port: 3200
    http_listen_conn_limit: 0
    grpc_listen_network: tcp
    grpc_listen_address: ""
    grpc_listen_port: 9095
    grpc_listen_conn_limit: 0
    tls_cipher_suites: ""
    tls_min_version: ""
    http_tls_config:
        cert_file: ""
        key_file: ""
        client_auth_type: ""
        client_ca_file: ""
    grpc_tls_config:
        cert_file: ""
        key_file: ""
        client_auth_type: ""
        client_ca_file: ""
    register_instrumentation: true
    graceful_shutdown_timeout: 30s
    http_server_read_timeout: 30s
    http_server_write_timeout: 30s
    http_server_idle_timeout: 2m0s
    grpc_server_max_recv_msg_size: 134217728
    grpc_server_max_send_msg_size: 134217728
    grpc_server_max_concurrent_streams: 100
    grpc_server_max_connection_idle: 2562047h47m16.854775807s
    grpc_server_max_connection_age: 2562047h47m16.854775807s
    grpc_server_max_connection_age_grace: 2562047h47m16.854775807s
    grpc_server_keepalive_time: 2h0m0s
    grpc_server_keepalive_timeout: 20s
    grpc_server_min_time_between_pings: 10s
    grpc_server_ping_without_stream_allowed: true
    log_format: logfmt
    log_level: info
    log_source_ips_enabled: false
    log_source_ips_header: ""
    log_source_ips_regex: ""
    log_request_at_info_level_enabled: false
    http_path_prefix: ""
distributor:
    ring:
        kvstore:
            store: memberlist
            prefix: collectors/
            consul:
                host: localhost:8500
                acl_token: ""
                http_client_timeout: 20s
                consistent_reads: false
                watch_rate_limit: 1
                watch_burst_size: 1
                cas_retry_delay: 1s
            etcd:
                endpoints: []
                dial_timeout: 10s
                max_retries: 10
                tls_enabled: false
                tls_cert_path: ""
                tls_key_path: ""
                tls_ca_path: ""
                tls_server_name: ""
                tls_insecure_skip_verify: false
                username: ""
                password: ""
            multi:
                primary: ""
                secondary: ""
                mirror_enabled: false
                mirror_timeout: 2s
        heartbeat_period: 5s
        heartbeat_timeout: 5m0s
        instance_id: compactor-6496d59455-4t4xc
        instance_interface_names:
            - eth0
            - en0
        instance_port: 0
        instance_addr: ""
    receivers: {}
    override_ring_key: distributor
    log_received_traces: false
    extend_writes: true
    search_tags_deny_list: []
ingester_client:
    pool_config:
        checkinterval: 15s
        healthcheckenabled: true
        healthchecktimeout: 1s
    remote_timeout: 5s
    grpc_client_config:
        max_recv_msg_size: 104857600
        max_send_msg_size: 104857600
        grpc_compression: snappy
        rate_limit: 0
        rate_limit_burst: 0
        backoff_on_ratelimits: false
        backoff_config:
            min_period: 100ms
            max_period: 10s
            max_retries: 10
        tls_enabled: false
        tls_cert_path: ""
        tls_key_path: ""
        tls_ca_path: ""
        tls_server_name: ""
        tls_insecure_skip_verify: false
metrics_generator_client:
    pool_config:
        checkinterval: 15s
        healthcheckenabled: true
        healthchecktimeout: 1s
    remote_timeout: 5s
    grpc_client_config:
        max_recv_msg_size: 104857600
        max_send_msg_size: 104857600
        grpc_compression: snappy
        rate_limit: 0
        rate_limit_burst: 0
        backoff_on_ratelimits: false
        backoff_config:
            min_period: 100ms
            max_period: 10s
            max_retries: 10
        tls_enabled: false
        tls_cert_path: ""
        tls_key_path: ""
        tls_ca_path: ""
        tls_server_name: ""
        tls_insecure_skip_verify: false
querier:
    search:
        query_timeout: 30s
        prefer_self: 2
        external_endpoints: []
        external_hedge_requests_at: 8s
        external_hedge_requests_up_to: 2
    query_timeout: 10s
    max_concurrent_queries: 5
    frontend_worker:
        frontend_address: ""
        dns_lookup_duration: 10s
        parallelism: 2
        match_max_concurrent: true
        id: ""
        grpc_client_config:
            max_recv_msg_size: 104857600
            max_send_msg_size: 134217728
            grpc_compression: gzip
            rate_limit: 0
            rate_limit_burst: 0
            backoff_on_ratelimits: false
            backoff_config:
                min_period: 100ms
                max_period: 1s
                max_retries: 5
            tls_enabled: false
            tls_cert_path: ""
            tls_key_path: ""
            tls_ca_path: ""
            tls_server_name: ""
            tls_insecure_skip_verify: false
    query_relevant_ingesters: false
query_frontend:
    max_outstanding_per_tenant: 100
    querier_forget_delay: 0s
    max_retries: 2
    search:
        concurrent_jobs: 50
        target_bytes_per_job: 10485760
        default_result_limit: 20
        max_result_limit: 0
        max_duration: 1h1m0s
        query_backend_after: 15m0s
        query_ingesters_until: 1h0m0s
    trace_by_id:
        query_shards: 20
        hedge_requests_at: 2s
        hedge_requests_up_to: 2
compactor:
    ring:
        kvstore:
            store: memberlist
            prefix: collectors/
            consul:
                host: localhost:8500
                acl_token: ""
                http_client_timeout: 20s
                consistent_reads: false
                watch_rate_limit: 1
                watch_burst_size: 1
                cas_retry_delay: 1s
            etcd:
                endpoints: []
                dial_timeout: 10s
                max_retries: 10
                tls_enabled: false
                tls_cert_path: ""
                tls_key_path: ""
                tls_ca_path: ""
                tls_server_name: ""
                tls_insecure_skip_verify: false
                username: ""
                password: ""
            multi:
                primary: ""
                secondary: ""
                mirror_enabled: false
                mirror_timeout: 2s
        heartbeat_period: 5s
        heartbeat_timeout: 1m0s
        wait_stability_min_duration: 1m0s
        wait_stability_max_duration: 5m0s
        instance_id: compactor-6496d59455-4t4xc
        instance_interface_names:
            - eth0
            - en0
        instance_port: 0
        instance_addr: ""
        wait_active_instance_timeout: 10m0s
    compaction:
        chunk_size_bytes: 10485760
        flush_size_bytes: 20971520
        compaction_window: 1h0m0s
        max_compaction_objects: 6000000
        max_block_bytes: 107374182400
        block_retention: 1h0m0s
        compacted_block_retention: 15m0s
        retention_concurrency: 10
        iterator_buffer_size: 1000
        max_time_per_tenant: 5m0s
        compaction_cycle: 30s
    override_ring_key: compactor
ingester:
    lifecycler:
        ring:
            kvstore:
                store: memberlist
                prefix: collectors/
                consul:
                    host: localhost:8500
                    acl_token: ""
                    http_client_timeout: 20s
                    consistent_reads: false
                    watch_rate_limit: 1
                    watch_burst_size: 1
                    cas_retry_delay: 1s
                etcd:
                    endpoints: []
                    dial_timeout: 10s
                    max_retries: 10
                    tls_enabled: false
                    tls_cert_path: ""
                    tls_key_path: ""
                    tls_ca_path: ""
                    tls_server_name: ""
                    tls_insecure_skip_verify: false
                    username: ""
                    password: ""
                multi:
                    primary: ""
                    secondary: ""
                    mirror_enabled: false
                    mirror_timeout: 2s
            heartbeat_timeout: 5m0s
            replication_factor: 3
            zone_awareness_enabled: false
            excluded_zones: ""
        num_tokens: 128
        heartbeat_period: 5s
        heartbeat_timeout: 1m0s
        observe_period: 0s
        join_after: 0s
        min_ready_duration: 15s
        interface_names:
            - eth0
        final_sleep: 0s
        tokens_file_path: ""
        availability_zone: ""
        unregister_on_shutdown: true
        readiness_check_ring_health: true
        address: ""
        port: 0
        id: compactor-6496d59455-4t4xc
    concurrent_flushes: 16
    flush_check_period: 10s
    flush_op_timeout: 5m0s
    trace_idle_period: 10s
    max_block_duration: 1h0m0s
    max_block_bytes: 1073741824
    complete_block_timeout: 15m0s
    override_ring_key: ring
    use_flatbuffer_search: false
metrics_generator:
    ring:
        kvstore:
            store: memberlist
            prefix: collectors/
            consul:
                host: localhost:8500
                acl_token: ""
                http_client_timeout: 20s
                consistent_reads: false
                watch_rate_limit: 1
                watch_burst_size: 1
                cas_retry_delay: 1s
            etcd:
                endpoints: []
                dial_timeout: 10s
                max_retries: 10
                tls_enabled: false
                tls_cert_path: ""
                tls_key_path: ""
                tls_ca_path: ""
                tls_server_name: ""
                tls_insecure_skip_verify: false
                username: ""
                password: ""
            multi:
                primary: ""
                secondary: ""
                mirror_enabled: false
                mirror_timeout: 2s
        heartbeat_period: 5s
        heartbeat_timeout: 1m0s
        instance_id: compactor-6496d59455-4t4xc
        instance_interface_names:
            - eth0
            - en0
        instance_addr: ""
    processor:
        service_graphs:
            wait: 10s
            max_items: 10000
            workers: 10
            histogram_buckets:
                - 0.1
                - 0.2
                - 0.4
                - 0.8
                - 1.6
                - 3.2
                - 6.4
                - 12.8
            dimensions: []
        span_metrics:
            histogram_buckets:
                - 0.002
                - 0.004
                - 0.008
                - 0.016
                - 0.032
                - 0.064
                - 0.128
                - 0.256
                - 0.512
                - 1.024
                - 2.048
                - 4.096
                - 8.192
                - 16.384
            dimensions: []
    registry:
        collection_interval: 15s
        stale_duration: 15m0s
    storage:
        path: ""
        wal:
            wal_segment_size: 134217728
            wal_compression: false
            stripe_size: 16384
            truncate_frequency: 2h0m0s
            min_wal_time: 300000
            max_wal_time: 14400000
            no_lockfile: false
        remote_write_flush_deadline: 1m0s
        remote_write:
            - url: http://kube-prometheus-stack-prometheus.monitoring.svc.cluster.local:9090/api/v1/write
              remote_timeout: 30s
              name: remote-write
              send_exemplars: true
              follow_redirects: true
              enable_http2: true
              queue_config:
                capacity: 2500
                max_shards: 200
                min_shards: 1
                max_samples_per_send: 500
                batch_send_deadline: 5s
                min_backoff: 30ms
                max_backoff: 5s
              metadata_config:
                send: true
                send_interval: 1m
                max_samples_per_send: 500
    metrics_ingestion_time_range_slack: 30s
storage:
    trace:
        pool:
            max_workers: 50
            queue_depth: 2000
        wal:
            path: /var/tempo/wal
            completedfilepath: /var/tempo/wal/completed
            blocksfilepath: /var/tempo/wal/blocks
            encoding: snappy
            search_encoding: none
            ingestion_time_range_slack: 2m0s
        block:
            index_downsample_bytes: 1048576
            index_page_size_bytes: 256000
            bloom_filter_false_positive: 0.01
            bloom_filter_shard_size_bytes: 102400
            version: vParquet
            encoding: zstd
            search_encoding: snappy
            search_page_size_bytes: 1048576
            row_group_size_bytes: 100000000
        search:
            chunk_size_bytes: 1000000
            prefetch_trace_count: 1000
            read_buffer_count: 8
            read_buffer_size_bytes: 4194304
            cache_control:
                footer: false
                column_index: false
                offset_index: false
        blocklist_poll: 5m0s
        blocklist_poll_concurrency: 50
        blocklist_poll_fallback: true
        blocklist_poll_tenant_index_builders: 2
        blocklist_poll_stale_tenant_index: 0s
        blocklist_poll_jitter_ms: 0
        backend: s3
        local:
            path: ""
        gcs:
            bucket_name: ""
            chunk_buffer_size: 10485760
            endpoint: ""
            hedge_requests_at: 0s
            hedge_requests_up_to: 2
            insecure: false
            object_cache_control: ""
            object_metadata: {}
        s3:
            bucket: tempo
            endpoint: minio.minio.svc.cluster.local:9000
            region: ""
            access_key: 6wnahMaygzx2GTgw
            secret_key: '********'
            insecure: true
            insecure_skip_verify: false
            part_size: 0
            hedge_requests_at: 0s
            hedge_requests_up_to: 2
            signature_v2: false
            forcepathstyle: false
            tags: {}
            storage_class: ""
            metadata: {}
        azure:
            storage-account-name: ""
            storage-account-key: ""
            use-managed-identity: false
            user-assigned-id: ""
            container-name: ""
            endpoint-suffix: blob.core.windows.net
            max-buffers: 4
            buffer-size: 3145728
            hedge-requests-at: 0s
            hedge-requests-up-to: 2
        cache: memcached
        cache_min_compaction_level: 0
        cache_max_block_age: 0s
        background_cache:
            writeback_goroutines: 10
            writeback_buffer: 10000
        memcached:
            host: memcached
            service: memcached-client
            addresses: ""
            timeout: 200ms
            max_idle_conns: 16
            max_item_size: 0
            update_interval: 1m0s
            consistent_hash: true
            circuit_breaker_consecutive_failures: 0
            circuit_breaker_timeout: 0s
            circuit_breaker_interval: 0s
            ttl: 0s
        redis: null
overrides:
    ingestion_rate_strategy: local
    ingestion_rate_limit_bytes: 15000000
    ingestion_burst_size_bytes: 20000000
    search_tags_allow_list: null
    max_traces_per_user: 10000
    max_global_traces_per_user: 0
    max_search_bytes_per_trace: 5000
    metrics_generator_ring_size: 0
    metrics_generator_processors: null
    metrics_generator_max_active_series: 0
    metrics_generator_collection_interval: 0s
    metrics_generator_disable_collection: false
    metrics_generator_forwarder_queue_size: 0
    metrics_generator_forwarder_workers: 0
    metrics_generator_processor_service_graphs_histogram_buckets: []
    metrics_generator_processor_service_graphs_dimensions: []
    metrics_generator_processor_span_metrics_histogram_buckets: []
    metrics_generator_processor_span_metrics_dimensions: []
    block_retention: 0s
    max_bytes_per_tag_values_query: 5000000
    max_search_duration: 0s
    max_bytes_per_trace: 5000000
    per_tenant_override_config: /overrides/overrides.yaml
    per_tenant_override_period: 10s
memberlist:
    node_name: ""
    randomize_node_name: true
    stream_timeout: 10s
    retransmit_factor: 2
    pull_push_interval: 30s
    gossip_interval: 1s
    gossip_nodes: 2
    gossip_to_dead_nodes_time: 30s
    dead_node_reclaim_time: 0s
    compression_enabled: false
    advertise_addr: ""
    advertise_port: 7946
    cluster_label: ""
    cluster_label_verification_disabled: false
    join_members:
        - gossip-ring.tempo.svc.cluster.local.:7946
    min_join_backoff: 1s
    max_join_backoff: 1m0s
    max_join_retries: 10
    abort_if_cluster_join_fails: false
    rejoin_interval: 0s
    left_ingesters_timeout: 5m0s
    leave_timeout: 20s
    message_history_buffer_bytes: 0
    bind_addr: []
    bind_port: 7946
    packet_dial_timeout: 2s
    packet_write_timeout: 5s
    tls_enabled: false
    tls_cert_path: ""
    tls_key_path: ""
    tls_ca_path: ""
    tls_server_name: ""
    tls_insecure_skip_verify: false
usage_report:
    reporting_enabled: true
    backoff:
        min_period: 100ms
        max_period: 10s
        max_retries: 0
```

**Comment by irizzant** [comment]
Hi, any update?

**Comment by ie-pham** [comment]
Hey there!

Looking over the logs, there were a few things we wanted to follow up on. We noticed this in the logs:

```
2022-10-10T15:20:39+02:00	level=info ts=2022-10-10T13:20:39.125831854Z caller=compactor.go:239 msg="flushed to block" bytes=4671170 objects=869 values=99487811
2022-10-10T15:18:16+02:00	level=info ts=2022-10-10T13:18:16.982679612Z caller=compactor.go:239 msg="flushed to block" bytes=4671170 objects=869 values=99487811
2022-10-10T15:22:51+02:00	level=info ts=2022-10-10T13:22:51.578900995Z caller=compactor.go:239 msg="flushed to block" bytes=4671170 objects=869 values=99487811
2022-10-10T15:52:05+02:00	level=info ts=2022-10-10T13:52:05.048697056Z caller=compactor.go:239 msg="flushed to block" bytes=5047128 objects=896 values=99901316
```

Are the logs from the session where the row_group_size was set to 100MB? If not, could you provide those logs? As this shows that it is still trying to flush less than 5MB to S3/Minio and this is what is trigger the "Your proposed upload is smaller than the minimum allowed object size" error. If the logs is from when the row_group_size is set to 100MB, would it be possible to increase the row_group_size again? It could be that your traces have a particular pattern that is exacerbating a known issue we have.

**Comment by irizzant** [comment]
Hi @ie-pham 

the logs are from the session where the row_group_size was 100MB, I'm going to try increasing the row_group_size to 200MB and let you know.

**Comment by irizzant** [comment]
Hi @ie-pham 

I'm using right now 300MB as row_group_size and I've also updated to tempo image:
grafana/tempo@sha256:87d49512e192b05d6c1896c34ac86f58fc9be6f5429921acb157659fc74628ea

So far so good, I don't see "Your proposed upload is smaller than the minimum allowed object size" anymore.
Also, as you can see from the screen below, the volume usage is being freed by the compactor.

What is the suggested way to proceed at this point?
Should I keep the huge row_group_size like this? Should I wait for a newer Tempo release?

Thanks

![image](https://user-images.githubusercontent.com/15982257/200031114-8f074c8f-218c-49ac-8c47-ed1c82ec86d4.png)

**Comment by joe-elliott** [comment]
hey @irizzant glad to hear things are heading the right direction here. I have just submitted a PR which makes `row_group_size_bytes` a lower bound for the row group size created (#1873).

we still don't have a direct way to calculate the number of bytes used, but at least this will tend to create larger rather than smaller row groups. we have also found that larger row groups tend to improve query performance and reduce ingester load.

**Comment by irizzant** [comment]
Hi @joe-elliott thanks for the updates, so you're suggesting to wait for https://github.com/grafana/tempo/pull/1873 to be merged and then start using `row_group_size_bytes` right?

**Comment by joe-elliott** [comment]
Yup! At that point I would remove your custom setting of 200MB and see if the default of 100MB works for you.

**Comment by irizzant** [comment]
Ok @joe-elliott thank you, I see that #1873 was merged when is it going to be released?

**Comment by joe-elliott** [comment]
This will be released with 2.0. We are very close. There is a final PR to stabilize and then it's only cleanup after that.

https://github.com/grafana/tempo/pull/1878

**Comment by irizzant** [comment]
Ok thank you I'm going to close this then
